flow/layers/backdrop_filter_layer.cc

```diff
@@ -9,6 +9,17 @@ namespace flutter {
 BackdropFilterLayer::BackdropFilterLayer(sk_sp<SkImageFilter> filter)
     : filter_(std::move(filter)) {}
 
+void BackdropFilterLayer::Preroll(PrerollContext* context,
```

Comment:
Why no let `Layer::Preroll` also return `bool` like the `LayerTree::Preroll` does? By returning bool, we can simplify most `Layer::Preroll` as

```
bool XyzLayer::Preroll(...) {
  bool children_needs_read_from_surface = ContainerLayer::Preroll(...);
  return this_layer_needs_read_from_surface() || !use_save_layer() && children_needs_read_from_surface;
}

bool ContainerLayer::Preroll(...) {
  bool children_needs_read_from_surface = false;
  for (child in children) {
    children_needs_read_from_surface ||= child->Preroll(...);
  }
  return children_needs_read_from_surface;
}
```

Reply:
Eventually Preroll will collect more information, such as the dirty regions. Even now the context is tracking multiple pieces of information that are modified and overridden on a layer by layer basis and so it seems that is the best way to collect any data that is negotiated between layers in the tree.